be/src/common/config.h

```diff
@@ -849,6 +849,8 @@ CONF_Bool(enable_simdjson_reader, "false");
 CONF_Int32(doris_remote_scanner_thread_pool_thread_num, "16");
 // number of s3 scanner thread pool queue size
 CONF_Int32(doris_remote_scanner_thread_pool_queue_size, "10240");
+// write offsets instead of length to speed up seek performance
+CONF_Bool(enable_write_array_offset, "false");
```

Comment:
1. After the data persist, we can not change this flag any more. If really needed, I think we should add some meta to distinguish it is offset or length inside segment file;
2. How about if we delete one row with ARRAY column?
3. Does clickhouse store ARRAY length or offset?

Reply:
1. the code provided both backward and forward compability, use `_length_iterator->get_current_page()->next_array_item_ordinal > 0` to distinguish whether we go the offset path else length path
2. I think delete row is not effected whether using offset or length, it's re compacted and written a new column?
3. clickhous store array offset

Reply:
I can add a seperate meta to distinguish them

Reply:
Maybe we should just turn if on by default?

Reply:
I think we should better choose one before doris1.2 release.
If we decide to choose one, we need more test cases.
This pr focus on random read of Array column, `offset version` is better than `length version` in this case.
Other cases, such as sequence read, we also need to compare.

Please add my wechat (doris_cambyzju).

Reply:
I will provide sequence read case too @cambyzju

Reply:
I had a conversation with @cambyzju  and decided to use  `offset version`, so this PR is incompatible with previous